Server/mods/deathmatch/logic/lua/CLuaFunctionParseHelpers.h

```diff
@@ -437,6 +441,24 @@ CPlayer* UserDataCast(CPlayer*, void* ptr, lua_State*)
     return (CPlayer*)pElement;
 }
 
+// CClient from CConsoleClient or a CPlayer
+template <class T>
+CClient* UserDataCast(CClient*, void* ptr, lua_State*)
+{
+    ElementID ID = TO_ELEMENTID(ptr);
+    CElement* pElement = CElementIDs::GetElement(ID);
+    if (!pElement || pElement->IsBeingDeleted() || (pElement->GetType() != CElement::PLAYER && pElement->GetType() != CElement::CONSOLE))
+        return nullptr;
+
+    CClient* pClient = nullptr;
+    if (pElement->GetType() == CElement::PLAYER)
+        pClient = (CClient*)reinterpret_cast<CPlayer*>(pElement);
```

Comment:
The `(CClient*)` cast shouldn't be neccessary, since `CPlayer` inherits from `CClient`

Reply:
clang is being a beetch https://github.com/multitheftauto/mtasa-blue/pull/1427#issuecomment-624122774